pkgs/coverage/bin/test_with_coverage.dart

```diff
@@ -93,13 +86,13 @@ ArgParser _createArgParser(CoverageOptions defaultOptions) => ArgParser()
       defaultsTo: defaultOptions.scopeOutput,
       help: 'restrict coverage results so that only scripts that start with '
           'the provided package path are considered. Defaults to the name of '
-          'the package under test.')
+          'the current package (including all subpackages, if this is a '
```

Comment:
I'm assuming one can still specify _which_ subpackages one want coverage for with `--scope-output` manually?

E.g.

* pkgs/
  * my_package/
    * lib/ <-- I want coverage for this.
    * test_data/
      * my_test_package/
        * lib/  <-- I don't want coverage for this

If I use `dart pub global run coverage:test_with_coverage --scope-output my_package -- pkgs/my_package/test` will that do what I want?

The test projects are usually part of the workspace w.r.t. analysis. However, they are not part of the packages we want code coverage stats for.

(I know it's not what this PR is doing, because it's automating the `--scope-output` if not provided. But maybe we can add a test for this use case. Fine to do that in a separate PR.)

Reply:
The scoping is based on the package URI:

```
package:my_package/my_package.dart
        \________/
          scope
```

`my_test_package` is a separate package with a different URI, so `dart pub global run coverage:test_with_coverage -- pkgs/my_package/test` should work. But yeah, you can also provide the scopes manually if you want.